sphinx/ext/intersphinx.py

```diff
@@ -271,237 +330,253 @@ def load_mappings(app: Sphinx) -> None:
-    else:
-        reftitle = _('(in %s)') % (proj,)
-    newnode = nodes.reference('', '', internal=False, refuri=uri, reftitle=reftitle)
-    if node.get('refexplicit'):
-        # use whatever title was given
-        newnode.append(contnode)
-    elif dispname == '-' or \
-            (domain.name == 'std' and node['reftype'] == 'keyword'):
-        # use whatever title was given, but strip prefix
-        title = contnode.astext()
-        if inv_name is not None and title.startswith(inv_name + ':'):
-            newnode.append(contnode.__class__(title[len(inv_name) + 1:],
-                                              title[len(inv_name) + 1:]))
-        else:
-            newnode.append(contnode)
-    else:
-        # else use the given display name (used for :ref:)
-        newnode.append(contnode.__class__(dispname, dispname))
-    return newnode
-
-
-def _resolve_reference_in_domain_by_target(
-        inv_name: str | None, inventory: Inventory,
-        domain: Domain, objtypes: Iterable[str],
-        target: str,
-        node: pending_xref, contnode: TextElement) -> nodes.reference | None:
-    for objtype in objtypes:
-        if objtype not in inventory:
-            # Continue if there's nothing of this kind in the inventory
-            continue
-
-        if target in inventory[objtype]:
-            # Case sensitive match, use it
-            data = inventory[objtype][target]
-        elif objtype in {'std:label', 'std:term'}:
-            # Some types require case insensitive matches:
-            # * 'term': https://github.com/sphinx-doc/sphinx/issues/9291
-            # * 'label': https://github.com/sphinx-doc/sphinx/issues/12008
-            target_lower = target.lower()
-            insensitive_matches = list(filter(lambda k: k.lower() == target_lower,
-                                              inventory[objtype].keys()))
-            if insensitive_matches:
-                data = inventory[objtype][insensitive_matches[0]]
-            else:
-                # No case insensitive match either, continue to the next candidate
-                continue
-        else:
-            # Could reach here if we're not a term but have a case insensitive match.
-            # This is a fix for terms specifically, but potentially should apply to
-            # other types.
-            continue
-        return _create_element_from_result(domain, inv_name, data, node, contnode)
-    return None
+                domain_entries = entries.setdefault(domain_name, {})
+                per_type = domain_entries.setdefault(object_type, {})
+                for object_name, object_data in inv_objects.items():
+                    item_set = per_type.setdefault(object_name, InventoryItemSet())
+                    item_set.append(inv_name, object_data)
+
+        # and then give the data to each domain
+        for domain_name, domain_entries in entries.items():
+            if _debug:
+                global _debug_indent
```

Comment:
Honestly, I don't want a global variable like that. I'd prefer having a logger.debug() call.